core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -394,8 +393,7 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
    * @param configs properties to validate for the IP
    */
   def validateIpConfig(ip: String, configs: Properties): Unit = {
-    if (ip != ConfigEntityName.Default && !Utils.validHostPattern(ip))
-      throw new AdminOperationException(s"IP $ip is not a valid address.")
+    DynamicConfig.Ip.validateIpOrHost(ip)
```

Comment:
It is a little weird to throw `InvalidRequestException` here because there is not request involved. Users of `AdminZkClient` may not expect this. I think that it would be better to throw a `IllegalArgumentException`. Then in the `AdminManager#alterClientQuotas`, we can catch it and transform it into `InvalidRequestException` by copying the internal message.